library/manageiq_alert.py

```diff
@@ -160,8 +167,9 @@ def alert_update_required(self, alert_id, description, expression, miq_entity, o
         except Exception as e:
             self.module.fail_json(msg="Failed to get alert {description} details. Error: {error}".format(description=description, error=e))
 
-        # remove None values from expression and options dicts
-        current_expression = {k: v for k, v in result['expression']['exp'].items() if v is not None}
+        # remove None values from expression and options dicts, if needed
+        if expression_type == 'miq_expression':
+            current_expression = {k: v for k, v in result['expression']['exp'].items() if v is not None}
```

Comment:
Done.

Also added a TODO comment to use the returned `expression_type`, once supported by ManangeIQ API.